internal/venice-consumer/src/main/java/com/linkedin/venice/pubsub/kafka/KafkaTopicPartition.java

```diff
@@ -0,0 +1,34 @@
+package com.linkedin.venice.pubsub.kafka;
+
+import com.linkedin.venice.pubsub.api.PubSubTopic;
+import com.linkedin.venice.pubsub.api.PubSubTopicPartition;
+import org.apache.kafka.common.TopicPartition;
+
+
+public class KafkaTopicPartition implements PubSubTopicPartition {
+  private final PubSubTopic topic;
+  private final TopicPartition topicPartition;
+
+  KafkaTopicPartition(PubSubTopic topic, TopicPartition topicPartition) {
+    this.topic = topic;
+    this.topicPartition = topicPartition;
+  }
+
+  KafkaTopicPartition(PubSubTopic topic, int partition) {
+    this(topic, new TopicPartition(topic.getName(), partition));
+  }
+
+  @Override
+  public PubSubTopic getPubSubTopic() {
+    return topic;
+  }
+
+  @Override
+  public int getPartitionNumber() {
+    return topicPartition.partition();
+  }
+
+  public TopicPartition getTopicPartition() {
```

Comment:
If you're planning on this class, let's not expose TopicPartition.

Reply:
I think `KafkaTopicPartition` is for kafka world and `TopicPartition` is an important term in kafka world, so this might simplify kafka related logic.

Reply:
> I think `KafkaTopicPartition` is for kafka world and `TopicPartition` is an important term in kafka world, so this might simplify kafka related logic.

I don't think thats a great defense of this.  Recall, we aspire to have the streaming solution under the hood to be pluggable (like with Pulsar as an example).  In our current interfaces we expose the notion or "partition" but not "topic".  I think it's ok to have topic in the name of the concretion, but maybe not the interface.

Reply:
mmmm though on second thought, glancing through the api spec for a few examples (Redpanda, Google pubsub, and even Pulsar) the usage of topic seems to have become common nomenclature for persistent streams.  Maybe this is ok.

Reply:
I think we can keep this at this moment, given this is the initial step.